nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/batch/handler/impl/DefaultBatchHandlerImpl.java

```diff
@@ -0,0 +1,114 @@
+package org.nuxeo.ecm.automation.server.jaxrs.batch.handler.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.Batch;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.BatchFileEntry;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.handler.AbstractBatchHandler;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.handler.BatchFileInfo;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.transientstore.api.TransientStore;
+import org.nuxeo.ecm.core.transientstore.api.TransientStoreService;
+import org.nuxeo.runtime.api.Framework;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class DefaultBatchHandlerImpl extends AbstractBatchHandler {
+
+    protected static final Log log = LogFactory.getLog(DefaultBatchHandlerImpl.class);
+
+    protected static final String BATCH_HANDLER_NAME = "default";
+
+    private static class Config {
+        public static final String TRANSIENT_STORE_NAME = "transientStore";
+    }
+
+    private TransientStore transientStore;
+    private TransientStoreService transientStoreService;
+
+    private String transientStoreName;
+
+    public DefaultBatchHandlerImpl() {
+        super(BATCH_HANDLER_NAME);
+    }
+
+    @Override public Batch newBatch() {
+        return initBatch();
+    }
+
+    protected Batch initBatch() {
+        return initBatch(null);
+    }
+
+    @Override public Batch newBatch(String providedId) {
+        return initBatch(providedId);
+    }
+
+    @Override public Batch getBatch(String batchId) {
+        TransientStore transientStore = getTransientStore();
+        Map<String, Serializable> batchEntryParams = transientStore.getParameters(batchId);
+
+        if (batchEntryParams == null) {
+            if (!hasBatch(batchId)) {
+                return null;
+            }
+            batchEntryParams = new HashMap<>();
+        }
+
+        String batchProvider = batchEntryParams.getOrDefault("provider", getName()).toString();
```

Comment:
Use a cast `(String)` instead of `toString` if you don't expect anything else than a `String` to be in this param.